lighthouse-core/lib/i18n/README.md

```diff
@@ -0,0 +1,283 @@
+
+#### String Translation in Google Translation Console
+
+* Googler is ready to kick off the TC pipeline again.
+
+```shell
+# collect UIStrings (to make sure everything is up to date)
+$ yarn i18n
+
+# Extract the CTC format files to translation console
+$ sh import-source-from-github.sh
+
+# Submit CL. Wait ~2 weeks for translations
+
+# Import the translated CTC format files to locales/ and bake them
+$ sh export-tc-dump-to-github.sh
+```
+
+#### String Replacement (runtime)
+
+See [Appendix A: How runtime string replacement works](#appendix)
+
+
+
+# Writing UIStrings with LHL
+
+❗TODO(exterkamp): explain all the comments and where they go/what they become.
+
+❗TODO(exterkamp): explain why we can't use some ICU like number formatting.
+
+We want to keep strings close to the code in which they are used so that developers can easily understand their context. We use `i18n.js` to extract the `UIStrings` strings from individual js files.
+
+LHL strings in each module are defined in a `UIStrings` object with the strings as its properties. JSDoc is sometimes used to provide additional information about each string.
+
+The LHL syntax is based primarily around the standardized [ICU message formatting](http://userguide.icu-project.org/formatparse/messages) syntax.
+
+
+### Basic example
+
+A simple string.
+
+```javascript
+const UIStrings = {
+  /** Imperative title of a Lighthouse audit that ... */
+  title: 'Minify CSS',
+};
+```
+
+For proper translation, **all** strings must be accompanied by a description, written as a preceeding comment.
+
+
+### Replacements and primitive formatting
+
+Replacements (aka substitutions) include string replacements like `{some_name}` number formatting like `{timeInMs, number, milliseconds}`.
+
+#### Direct ICU replacement
+
+`{some_name}` is called _Direct ICU_ since the replacement is a direct substitution of ICU with a variable and uses no custom formatting. This is simply a direct replacement of text into a string. Often used for proper nouns, code, or other text that is dynamic and added at runtime.
+
+ICU replacements must use a JSDoc-like syntax to specify an example for direct ICU replacements:
+
+*   To specify the description, use
+    * `@description <description>`
+*   To specify an example for an ICU replacement, use
+    * `@example {<ICU variable name>} <example for ICU replacement>`
+
+```javascript
+const UIStrings = {
+    /**
+      * @description Error message explaining ...
+      * @example {errorCode} NO_SPEEDLINE_FRAMES
+      */
+    didntCollectScreenshots: `Chrome didn't .... ({errorCode})`,
+};
+```
+
+#### Complex ICU replacement
+
+`{timeInMs, number, milliseconds}` is called _Complex ICU_ since the replacement is for complex numbers and uses the custom formatters in Lighthouse. The supported complex ICU formats are: `milliseconds`, `seconds`, `bytes`, `percent`, and `extendedPercent`.
```

Comment:
```suggestion
`{timeInMs, number, milliseconds}` is called _Complex ICU_ since the replacement is for numbers and other complex replacements that use the custom formatters in Lighthouse. The supported complex ICU formats are: `milliseconds`, `seconds`, `bytes`, `percent`, and `extendedPercent`.
```

Reply:
not attached to the specific language but `complex numbers` is it's own thing so would prefer to just leave it at complex replacements ;)